### PR TITLE
Add functionality for FMWPA - number of abondoned wells

### PR DESCRIPTION
**Pull request by joakim-hove**
Support for summary variable FMWPA - number of abondoned wells.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by bska** [comment]
I have a few remarks:

 1. If we're calculating this at the field level we should support the group level too.
 2. Do we really want to base this count directly on `SCHEDULE` data rather than the dynamic state in the simulator?
 3. More of concern about efficiency: The N-squared nature of `Schedule::abandoned_wells()` looks a bit concerning.  Is there no other way of deriving this count if it is indeed the count we actually want?

**Comment by joakim-hove** [comment]
> 1. If we're calculating this at the field level we should support the group level too.

I don't know if `GMWPA`is a group property at all - but in that case we could add that as well; it is not requested here and now though - so I would prefer to complete and merge this now.

> 2. Do we really want to base this count directly on SCHEDULE data rather than the dynamic state in the simulator? 

No - I agree it should be based on the dynamic state from the simulator; I'll rewrite.

> ... The N-squared nature of Schedule::abandoned_wells() looks a bit concerning. Is there no other way of deriving this count if it is indeed the count we actually want?

To be honest I consider this a weird feature which does not deserve much very much love and attention; but anyway - I'll see if I can find a better way.

**Comment by bska** [comment]
> > 1. If we're calculating this at the field level we should support the group level too.
> 
> I don't know if `GMWPA` is a group property at all - but in that case we could add that as well;

`GMWPA` is indeed a group-level summary property.

> it is not requested here and now though - so I would prefer to complete and merge this now.

Fair enough.

> > 3. The N-squared nature of `Schedule::abandoned_wells()` looks a bit concerning. Is there no other way of deriving this count if it is indeed the count we actually want?
> 
> To be honest I consider this a weird feature which does not deserve much very much love and attention; but anyway - I'll see if I can find a better way.

One way might be to add something along the lines of a "well first opened" field, representing the report step at which the well first opened, either to the `Well` object or to the `Schedule` container.  Then we'd be able to answer the question by comparing `report_step` to the candidate report step.

**Comment by joakim-hove** [comment]
> One way might be to add something along the lines of a "well first opened" field

yes - that was roughly my plan.

**Comment by joakim-hove** [comment]
OK - this has been completely reworked.

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -682,6 +682,23 @@ inline quantity injection_history( const fn_args& args ) {
...
+
+    for (const auto sched_well : args.schedule_wells) {
+        if (sched_well.hasProduced()) {
+            const auto& well_name = sched_well.name();
+            if (args.wells.count(well_name)) {
+                auto output_well = args.wells.at( sched_well.name() );
+                if (!output_well.flowing())
+                    count++;
+            }
+        }
```
Comment (bska) [inline src/opm/output/eclipse/Summary.cpp:696]: Technically, this looks up the well twice.  It's not a huge cost, but we could consider writing this loop body as
```C++
if (!sched_well.hasProduced())
    continue;

auto xwPos = args.wells.find(sched_well.name());
if (xwPos == args.wells.end())
    continue;

count += !xwPos->second.flowing();
```

**Review by bska — COMMENTED** [review]
Save for one minor detail in the implementation of `abandoned_wells()`, I like this new version very much.

File: src/opm/output/eclipse/Summary.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/output/eclipse/Summary.cpp]: OK - I have rewritten to use an iterator.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by bska** [comment]
Build check green.  I'll merge into master.

**Pull request merged**